gst-libs/mfx/gstmfxencoder.c

```diff
@@ -406,6 +406,97 @@ gst_mfx_encoder_properties_get_default (const GstMfxEncoderClass * klass)
   return props;
 }
 
+static void
+gst_mfx_extVideoSignal_from_colorimetry(GstVideoColorimetry *colorimetry, mfxExtVideoSignalInfo *extVideo) {
+	memset(extVideo, 0, sizeof(mfxExtVideoSignalInfo));
+	extVideo->Header.BufferId = MFX_EXTBUFF_VIDEO_SIGNAL_INFO;
+	extVideo->Header.BufferSz = sizeof(mfxExtVideoSignalInfo);
+	extVideo->VideoFormat = 5; //  unspecified
+	switch (colorimetry->range)
+	{
+	case GST_VIDEO_COLOR_RANGE_16_235:
+		printf("\nUsing small range\n");
+		extVideo->VideoFullRange = 0; // not full range
+		break;
+	case GST_VIDEO_COLOR_RANGE_0_255:
+		printf("\nUsing full range\n");
+		extVideo->VideoFullRange = 1; // full range
+		break;
+	default:
+		printf("\nUsing unknown range\n");
+		extVideo->VideoFullRange = 0;
+		break;
+	}
+	extVideo->ColourDescriptionPresent = 1; // Do write this to the bitstream
+	switch (colorimetry->primaries) {
+	case GST_VIDEO_COLOR_PRIMARIES_BT709:
+		extVideo->ColourPrimaries = 1; // 709
+		break;
+	case GST_VIDEO_COLOR_PRIMARIES_BT470M:
+		extVideo->ColourPrimaries = 4; // 470M
+		break;
+	case GST_VIDEO_COLOR_PRIMARIES_BT470BG:
+		extVideo->ColourPrimaries = 5; // 470BG
+		break;
+	case GST_VIDEO_COLOR_PRIMARIES_SMPTE170M:
+		extVideo->ColourPrimaries = 6; // SMPTE170M
+		break;
+	case GST_VIDEO_COLOR_PRIMARIES_SMPTE240M:
+		extVideo->ColourPrimaries = 7; // SMPTE240M
+		break;
+	case GST_VIDEO_COLOR_PRIMARIES_FILM:
+		extVideo->ColourPrimaries = 8; // Generic film
+		break;
+	default:
+		extVideo->ColourPrimaries = 2; // Unspecified
+	}
+	switch (colorimetry->transfer) {
+	case GST_VIDEO_TRANSFER_BT709:
+		extVideo->TransferCharacteristics = 1;// BT709
```

Comment:
MFX_TRANSFERMATRIX_BT709 ?

Reply:
Maybe wrong. These enumerators may have been intended for mfxExtVPPVideoSignalInfo.

Reply:
I believe the numbers are supposed to be from Annex E ISO*/IEC* 14496-10 http://www.staroceans.org/e-book/ISO-14496-10.pdf

Reply:
You might want to add bt. 2020 support..
https://sourceforge.net/p/mediainfo/feature-requests/508/